src/lib/SCEditor.js

```diff
@@ -379,6 +380,56 @@ export default function SCEditor(original, userOptions) {
 	// Don't deep extend emoticons (fixes #565)
 	base.opts.emoticons = userOptions.emoticons || defaultOptions.emoticons;
 
+	if (!Array.isArray(options.allowedIframeUrls)) {
+		options.allowedIframeUrls = [];
+	}
+	options.allowedIframeUrls.push('https://www.youtube-nocookie.com/embed/');
+
+	// Create new instance of DOMPurify for each editor instance so can
+	// have different allowed iframe URLs
+	// eslint-disable-next-line new-cap
+	var domPurify = DOMPurify();
+
+	// Allow iframes for things like YouTube, see:
+	// https://github.com/cure53/DOMPurify/issues/340#issuecomment-670758980
+	domPurify.addHook('uponSanitizeElement', function (node, data) {
+		var allowedUrls = options.allowedIframeUrls;
+
+		if (data.tagName === 'iframe') {
+			var src = node.getAttribute('src') || '';
+
+			for (var i = 0; i < allowedUrls.length; i++) {
+				var url = allowedUrls[i];
+
+				if (utils.isString(url) && src.substr(0, url.length) === url) {
+					return;
+				}
+
+				// Handle regex
+				if (url.test && url.test(src)) {
+					return;
+				}
+			}
+
+			// No match so remove
+			dom.remove(node);
+		}
+	});
+
+	/**
+	 * Sanitize HTML to avoid XSS
+	 *
+	 * @param {string} html
+	 * @return {string} html
+	 * @private
+	 */
+	function sanitize(html) {
+		return domPurify.sanitize(html, {
+			ADD_TAGS: ['iframe'],
+			ADD_ATTR: ['allowfullscreen', 'frameborder']
```

Comment:
Can these be options? Example: We use custom formats for our links which use target=blank.

Reply:
Ideally it should allow everything that's safe by default so configuring wouldn't be needed. I'm surprised dompurify is stripping target=blank, I know it needs rel=noopener to be fully secure in older browsers but even with that it's being stripped.

Reply:
Just realised IE doesn't support rel=noopener so that might explain it

A quick Google suggests adding `rel="noreferrer"` will fix the issue in IE and from testing it appears to work but according to [caniuse](https://caniuse.com/rel-noreferrer) it's only supported in IE 11 in Windows 10 creators update.

Possible solutions I can think of to allow target=blank safely: 

 * Do nothing except require rel=noopener and leave it to the page containing the editor to intercept links and make them safe in IE and Edge. Any page that supports IE and needs the extra security will probably have some way of dealing with this although if that was to strip target=blank from user input it would still be an issue in the editor
 * Add some code to intercept all target=blank links in IE and open them safely using JS 
 * Drop IE WYSIWYG support unless it supports noreferrer (not sure how easy this would be to detect)
 * Drop IE and pre-Blink Edge WYSIWYG support altogether (simplifies the codebase a lot)
 * Don't support target=blank and leave it to formats to add a data attribute specifying if a link is target=blank and adding it back when converting to source

Reply:
I've been thinking about this a bit more. As the point of this is to prevent security issues the first option is probably not a good idea and for compatibility not allowing target=blank wouldn't be good either.

Since this is already going to be a backwards incompatible update it might make sense to just drop IE and legacy Edge WYSIWYG support. A lot of sites seem to be dropping IE support including MS which will be dropping it from 365 and teams in August. IE support will be dropped eventually and doing so now would combine two non-backwards compatible updates in one.

If it was dropped then target=blank can be allowed safely with rel=noopener automatically added if missing. Just depends how important IE WYSIWYG support is.

Reply:
I looked at the source code for don-purify and saw some es6 features like arrow functions and the spread (lol funny name) operator. The transpile it with babel to support ie. This PR includes it wholesale, so no babel. (Explains the filesize difference between their build and what you reported above)

I'm  glad that Microsoft is going to finally drop IE soon. Disappointed that it took so long. Maybe now any stragglers left be a fire lit under them..

I fully support dropping support for IE and pre-chromium edge. This will help open the door to new  JS features, such as  classes.

Built-in formats do not use target=blank so support for that should be left to a plugin.

Reply:
It's not obvious from the source code but the dompurify NPM package actually contains the transpiled code so this PR shouldn't be including any ES6 features, the exact file this PR should be including is: https://github.com/cure53/DOMPurify/blob/main/dist/purify.es.js

Being able to use modern features, especially default parameters, const/let and arrow functions would definitely be nice. UglifyJS only has limited support for es6 features but I think it does include those.

> Built-in formats do not use target=blank so support for that should be left to a plugin.

If it wasn't such a common attribute I'd agree but as it is and by dropping IE support it can be allowed safely it might be worth including. Still not sure though.

Reply:
I wouldn't drop pre-chrome edge.  But IE makes sense.  Lots of enterprises are still on older LTS versions of Windows that have pre-chrome edge.  Some still have IE, but I think its inside that gap of minimal usage over effort required to maintain it.

Reply:
Do you know how long LTS for legacy Edge will last? If it's a long time it might be worth keeping but I can't seem to find anything that says. For regular users support will stop March and I think they will be auto upgraded to the new version in April but I can't find anything about extended support.

I know IE 11 has extended support available which will last until ~2025 but given MS are dropping it from 365 and a few other big sites either have or are going to drop support for it a lot of the web will stop working soon so probably safe to drop that.

Reply:
Server 2016, 2019 and Windows LTSC 2015/2016/2019 shipped with legacy edge.  But they can be upgraded to Chrome Edge.
https://support.microsoft.com/en-us/lifecycle/search?alpha=Windows%20Server%202016

2016 Server is support until 2027 it seems.  I would personally target 2025 to 2027 being when dropping pre-chrome edge would be safe as most enterprises will have moved off it... Hopefully.  I wouldn't worry too much about 2019 since it only adds a few more years but the target range of users will be very small.

I wouldn't support IE at all.  It may still ship with Windows, but its disabled by default.  No sensible admin would use IE11 with its lack of proper support for lots of modern tech.  If it works, its a best effort from the very few APIs it has.

Reply:
Is legacy Edge definitely included in LTSC? That page does show an end date for legacy Edge support but doesn't for the versions of IE that are still supported. It seems odd to show an end date for Edge if it is still supported but I can't find anything that says one way or the other.

Reply:
I've just found a [Windows IT Pro Blog post](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/ltsc-what-is-it-and-when-should-it-be-used/ba-p/293181) which says Edge isn't included in Windows 10 Enterprise LTSC so it might not be included in LTSC? I can't tell for sure as it doesn't say about other versions and I can't find anything other documentation about it.

Looking at statcounter usage though it's currently at 0.08% and falling and as it isn't included with Windows 10 Enterprise I'm leaning towards it being safe to drop. It will fallback to a plain a textarea so still usable. If you really need  WYSIYWG support for it I can keep it.

Reply:
Ok, I don't run LTSC servers at my work, so I guess bad info.  I do run 2019 DC servers though and they have Pre-Chrome Edge.  Besides our Remote Desktop servers where we have upgraded them to Chrome Edge.  Not sure how many enterprises do like we do though.

How much effort is it taking to do pre-chrome edge?  It had a lot of HTML5 features we use here.  So from the caniuse, it seems there is no effort to support any version of Edge.  So it seems safe to just drop IE.  If Edge works, it works.

Reply:
The lack of documentation available from Microsoft about legacy Edge support is surprising.

I did find [this page](https://docs.microsoft.com/en-us/microsoft-edge/deploy/about-microsoft-edge) yesterday which says Edge isn't included in LTSB versions, including server 2016, but that's all the official information I can find. I'm a bit confused as I've definitely seen it used on server 2016 but maybe that wasn't the LTS version? As far as I can tell it looks like support might actually end in March for everyone.

Supporting Edge requires a couple of fixes in the code for some quirks around contentEditable and to make it use the fallback clipboard API support. The main issue with it is testing as after IE 11, it has the most problems.

Like I say if you need support for it I can keep it but I'm reluctant to as the market share is very small and it looks like it isn't going to be supported soon based on the little information I can find.

It should still work in source mode if runWithoutWysiwygSupport is enabled and will fallback to a plain textarea if not so won't make anything unusable if not supported.

Reply:
Well then, I retract my statement on Edge.  Lets wait until we see reports about support or somebody wants to make the changes to introduce the support.